### ISSUE TITLE
How do I bootstrap packages available only within pip for running on Batch 

### ISSUE THREAD

**Issue by dibgerge** [Netflix/metaflow#395]
Running on AWS I would usually define a step:

```
@batch(cpu=1, memory=5000)
@conda(libraries={'pandas': '1'})
@step
def hello(self):
    do stuff...
```

However, I am working with deeplearning libraries (MXnet/tensorflow/pytorch), and they are not particularly up to date on conda and it is best to install with pip.

How do I define pip dependancies for this step?

**Comment by dibgerge** [comment]
I have been looking at the code, and it looks like the part of code where a Conda environment is created before it is uploaded to S3 is in `plugins.conda.conda`, `Conda._create` method:

```python
    def _create(self,
                env_id,
                deps,
                explicit=False,
                architecture=None,
                disable_safety_checks=False):
        cmd = ['create', '--yes', '--no-default-packages',
               '--name', env_id, '--quiet']
        if explicit:
            cmd.append('--no-deps')
        cmd.extend(deps)
        self._call_conda(cmd,
                         architecture=architecture,
                         disable_safety_checks=disable_safety_checks)
```

This directly conda, and there is no way to "inject" a pip package for installation. Is there some way to do this which I am missing?

**Comment by savingoyal** [comment]
@dibgerge There is some discussion on that topic in #24. You can install your pip packages in your step by executing - `subprocess.run([sys.executable, '-m', 'pip', 'install', '--quiet', ...])`.

**Comment by dibgerge** [comment]
Thanks, this solved it! I didn't see the topic. I was looking at the [docs](https://docs.metaflow.org/metaflow/dependencies), and I used the command  there: `os.system('pip install my_package')`, which was invoking the wrong `python` interpreter in the container. Didn't occur to me to use `sys.executable`. 

Are there plans to integrate this to have something like this instead: 

`@conda(libraries={...}, pip={...})`

so that pip libraries can be cached along with the conda libraries in s3?

**Comment by savingoyal** [comment]
@dibgerge Yes, we are going to look into that, especially after some of the [more recent changes in `pip`](http://pyfound.blogspot.com/2020/11/pip-20-3-new-resolver.html).

**Comment by tuulos** [comment]
consolidating all `pip` requests in this issue https://github.com/Netflix/metaflow/issues/24
